### PR TITLE
bug-fix\clear-matrix

### LINKED ISSUE #1129: font.print should print sentence on new line
image

```
expected output
hi
ok

```
we need to clear the letters matrix and bring the column pointer to zero after printing a sentence.

### PR DESCRIPTION
**Pull request by nakul-py**
*After printing a sentence i clear the letters matrix and reset the column pointer to zero .*

*OUTPUT*

Screenshot from 2024-10-10 10-47-48

closes #1129

### CODE REVIEW

**Comment by codewithnick** [comment]
@AnshumanMahato can you please verify this

**Comment by nakul-py** [comment]
@codewithnick will @AnshumanMahato review this quick?

**Comment by AnshumanMahato** [comment]
Give me some time... will do this by EOD

File: Fonts/fonts.h [hunk 1]
```diff
@@ -82,6 +82,16 @@ class Fonts
         }
     }
 
+    void resetMatrix()
+    {
+        letters.clear();
+        letters.reserve(def_rows);
+        for(int i = 0; i < def_rows; i++) {
```
Comment (AnshumanMahato) [inline Fonts/fonts.h:89]: I think everything seems good. Just change this `int i` to `unsigned int i`. We are getting some warnings due to a type mismatch. Also, fix any merge conflicts.

**Review by AnshumanMahato — CHANGES_REQUESTED** [review]

File: Fonts/fonts.h (hunk 1 above)
Comment (nakul-py) [reply Fonts/fonts.h]: @AnshumanMahato done

**Review by AnshumanMahato — APPROVED** [review]
Cool. Everything looks good. Thanks for contributing ✌️🙂

**Pull request merged**